includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -197,21 +191,16 @@ protected function update_post_meta( &$order, $force = false ) {
 		);
 
 		foreach ( $address_props as $props_key => $props ) {
-			foreach ( $props as $meta_key => $prop ) {
-				$prop_key = substr( $prop, 8 );
-				if ( ! $force && ( ! isset( $changed_props[ $props_key ] ) || ! array_key_exists( $prop_key, $changed_props[ $props_key ] ) ) ) {
-					continue;
-				}
+			$props_to_update = $this->get_props_to_update( $order, $props );
+			foreach ( $props_to_update as $meta_key => $prop ) {
 				$value = $order->{"get_$prop"}( 'edit' );
-
-				if ( '' !== $value ? update_post_meta( $order->get_id(), $meta_key, $value ) : delete_post_meta( $order->get_id(), $meta_key ) ) {
```

Comment:
Yeah there are a few places where the meta is deleted like that if the value is empty -- which would cause it to be reinserted as empty and deleted again in the future.